### ISSUE TITLE
Parse Error When Using Concatenation In Property Declaration

### ISSUE THREAD

**Issue by quantumpacket** [phpDocumentor/phpDocumentor2#1705]
The below code which is valid since PHP 5.6 produces the following error when generating documentation:

> Parse Error: Syntax error, unexpected '.', expecting ']' on line x  No summary was found for this file

```php
    /** @var array Default HTTP options */
    private $httpOptions = [
        'allow_redirects' => false,
        'decode_content'  => 'gzip',
        'headers'         => [
            'Accept-Encoding' => 'gzip',
            'Accept'          => 'application/json',
            'User-Agent'      => 'FooBar Client ' . Client::VERSION
                . ' (https://example.com/)'
        ]
    ];
```

**PHP:** 7.0.1
**PhpDoc:** dev-master

**Comment by stobbsm** [comment]
I'm having this issue as well. Documentation does not get generated for the file, which is quite important for my project.

PHP7 gives no problems, and neither does 5.6 when I tested my code there.

**Comment by quantumpacket** [comment]
What version of PHPdoc are you using? For me using *dev-master* it errors for PHP7+ and passed for PHP5.6. If PHP7 is not causing an issue for you then maybe the PHP version is not the culprit.

**Comment by jaapio** [comment]
We are using https://github.com/nikic/PHP-Parser version 0.9 to parse the phpcode. Which does support php <= 5.5.  I created a pr to update the parser to version ^1.0 https://github.com/phpDocumentor/Reflection/pull/93. Until this isn't merged any php 5.5+ syntax won't be supported.

On the other hand phpdocumentor 2 will not official have support for php7. This is scheduled for phpdoc v3. which is currently developed. We are also investigating what the effort will be to make phpdocumentor work on php7. We expect it to work on php7 if no php5.5+ syntax is used.

**Comment by jaapio** [comment]
releated to #1701
